dev/findings/W-32-variables-fanin.md

```
@@ -0,0 +1,67 @@
+status: draft
+upstream_issues:
+  - ghanse/wkmigrate#27
+---
+
+# W-32 — `@variables(...)` producer inside multi-activity ForEach: best-effort task key fails at runtime
+
+## Problem
+
+Step-3 audit (wkmigrate `6e0a3f5`, authored as part of CRP-11) documented a known limitation: when a `SetVariable` producer lives inside a multi-activity `ForEach` body and its consumer (an `IfCondition` referencing `@variables('X')`) sits at the outer scope, wkmigrate emits a **best-effort task key** that does not exist at Databricks runtime.
+
+```
```

markdownlint-cli2 (0.22.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)